### DISCUSSION TITLE
Specify another list in defaults

### DISCUSSION THREAD

**Discussion by pamparana34** [facebookresearch/hydra#2271] [Q&A] [answered]
I am trying to create a config as follows:

I have some optimiser config as:

``` 
# adam.yaml
_target_: torch.optim.Adam
lr: 1e-4
weight_decay: 0.0
```

And I have a few other ones as well like `SGD`

Now, my model config has something like:

```
_target_: models.SomeModel
defaults:
  - optimizer: [adam, SGD]
 ```

Now, when I do the it does not seem to like the fact that this is a list and only passes the first item to the underlying object. How can I pass a list in this case?

**Comment by Jasha10** [comment, marked as answer]
Adding an element `- optimizer: [adam, SGD]` to your defaults list will cause the adam and SGD configs to be *merged*. See [my comment here](https://github.com/facebookresearch/hydra/issues/1939#issuecomment-1035395006) for a tip on how to compose a ListConfig using the defaults list.
